core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -566,9 +566,11 @@ import scala.util.control.NonFatal
   private def commitAggregatedOffsets(): Unit = if (commitMaps.nonEmpty && !rebalanceInProgress) {
     val aggregatedOffsets = aggregateOffsets(commitMaps)
     commitRefreshing.add(aggregatedOffsets)
-    commit(aggregatedOffsets, commitSenders)
+    val replyTo = commitSenders
+    // flush the data before calling `consumer.commitAsync` which might call the callback synchronously
```

Comment:
Really?  That's counter intuitive based on the method name.  Good to know.

Reply:
I didn't expect it to. But in one of my manual tests it appeared as if it did and when I moved these it worked...

Reply:
It should really be the thread calling `poll` which executes the callback.